pkg/scaffold/controller/controllertest.go

```diff
@@ -59,17 +61,26 @@ func (a *Test) GetInput() (input.Input, error) {
 		"rbac.authorization":    "k8s.io",
 		"storage":               "k8s.io",
 	}
-	if _, found := coreGroups[a.Resource.Group]; found {
-		a.ResourcePackage = path.Join("k8s.io", "api")
-	} else {
-		a.ResourcePackage = path.Join(a.Repo, "pkg", "apis")
-	}
 
+	a.bindResource(coreGroups)
 	a.TemplateBody = controllerTestTemplate
 	a.Input.IfExistsAction = input.Error
 	return a.Input, nil
 }
 
+func (a *Test) bindResource(coreGroups map[string]string) {
+	resourcePath := filepath.Join("pkg", "apis", a.Resource.Group, a.Resource.Version,
+		fmt.Sprintf("%s_types.go", strings.ToLower(a.Resource.Kind)))
+	if _, err := os.Stat(resourcePath); os.IsNotExist(err) {
+		if _, found := coreGroups[a.Resource.Group]; found {
+			a.ResourcePackage = path.Join("k8s.io", "api")
+			return
+		}
+		// TODO: need to support '--resource-pkg-path' flag for specifying resourcePath
+	}
+	a.ResourcePackage = path.Join(a.Repo, "pkg", "apis")
+}
+
 var controllerTestTemplate = `{{ .Boilerplate }}
 
 package {{ lower .Resource.Kind }}
```

Comment:
We should add two tests by modifying the `generate_golden.sh` by creating controller for existing type under `pkg/apis` which matches the group with core and second one with controller-only for core resource.

Reply:
No problem. Added to test.